internal/rm/devices.go

```diff
@@ -150,6 +159,33 @@ func (ds Devices) Difference(ods Devices) Devices {
 	return res
 }
 
+// FilterByIDOrIndex returns the subset of devices matching the selected devices but not in excluded devices.
+// Basically, this is the combination of the above Subset and Difference methods.
+// It accepts inputs with the device format as ID(s) and Index(es).
+func (ds Devices) FilterByIDOrIndex(selected []string, excluded []string) Devices {
+	filterFunc := func(ds Devices, filter []string) Devices {
+		res := make(Devices)
+		for _, item := range filter {
+			if d := ds.GetByIDOrIndex(item); d != nil {
+				res[d.ID] = d
+			}
+		}
+		return res
+	}
+
+	res := ds
+	// Skip on the edge cases when seleted/excluded is an empty slice ([]string{""})
+	if !(len(selected) == 1 && selected[0] == "") {
```

Comment:
Not 100% sure but looks like docs are not covering this?

Reply:
How about this?

```suggestion

	// Only perform filtering when the "selected" is not an empty slice ([]string{""})
	if !(len(selected) == 1 && selected[0] == "") {
		f := filterFunc(ds, selected)
		res = res.Subset(f.GetIDs())
	}

	// Only perform filtering when the "excluded" is not an empty slice ([]string{""})
	if !(len(excluded) == 1 && excluded[0] == "") {
		f := filterFunc(ds, excluded)
		res = res.Difference(f)
	}


```